include/treelite/tree.h

```diff
@@ -474,12 +488,25 @@ class Model {
   /*! \brief disable copy; use default move */
   Model() = default;
   virtual ~Model() = default;
-  inline static std::unique_ptr<Model> Create();
   Model(const Model&) = delete;
   Model& operator=(const Model&) = delete;
   Model(Model&&) = default;
   Model& operator=(Model&&) = default;
 
+  template <typename ThresholdType, typename LeafOutputType>
+  inline static std::unique_ptr<Model> Create();
+  inline static std::unique_ptr<Model> Create(TypeInfo threshold_type, TypeInfo leaf_output_type);
+  inline TypeInfo GetThresholdType() const {
+    return threshold_type_;
+  }
+  inline TypeInfo GetLeafOutputType() const {
+    return leaf_output_type_;
+  }
+  template <typename Func>
+  inline auto Dispatch(Func func);
```

Comment:
Are these `Dispatch()` functions used anywhere?

Reply:
Yes, it's used in the code generator (excluded from this PR). See #196 to search for actual uses.

Reply:
Here is an example: https://github.com/dmlc/treelite/pull/196/files#diff-e85efd30a12be68c2d66cb6a7b674d48R124